File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* removeNthFromEnd(struct ListNode* head, int n) {
    struct ListNode* temp=head;
    struct ListNode* prev=NULL;
    if(head==NULL){
        return head;
    }
    if(head->next==NULL && n==1){
        return NULL;
    }
    int count=0;
    while(temp!=NULL){
        count++;
        temp=temp->next;
    }
    temp=head;
    if(n==count && temp->next!=NULL){
        struct ListNode* temp1=head;
        head=head->next;
        free(temp1);
        return head;
    }
    int i=0;
    temp=head;
    while(i<count-n && temp!=NULL){
        prev=temp;
        temp=temp->next;
        i++;
    }
    if(temp==NULL || temp->next==NULL){
       prev->next=NULL;
    }
    else{
        prev->next=temp->next;
        free(temp);
    }
    return head;
}

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def removeNthFromEnd(self, head, n):
        temp=head
        count=0
        while temp is not None:
            temp=temp.next
            count+=1
        req=count-n
        temp=head
        prev=None
        if req==0:
            return head.next
        while req!=0:
            prev=temp
            temp=temp.next
            req-=1
        prev.next=temp.next
        del(temp)
        return head
        